### PR TITLE
Simplify text system queries

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Both `measure_text_system` and `text_system` have `ParamSet` parameters each with three text queries. This complexity doesn't seem necessary and the `ParamSet` queries can be replaced with a single combined query. 

## Solution

Replace the `ParamSet`s in `measure_text_system` and `text_system` with single combined queries.

---

## Changelog

* Replaced the `measure_texture_system` and `text_system` `ParamSet` parameters with single combined queries.

### CODE REVIEW

**Comment by ickshonpe** [comment]
Seems to give a marginal performance improvement as well:

```cargo run --example many_buttons --profile stress-test --features trace_tracy -- recompute-text```

text_system-many_buttons-stress-test-a616fa8f70054d1c504bc-vs-baece543878

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ui/src/widget/text.rs [hunk 1]
```diff
@@ -79,27 +77,26 @@ pub fn measure_text_system(
...
+                text_queue.push(entity);
             }
         }
     } else {
         // If the scale factor has changed, queue all text
-        for entity in text_queries.p1().iter() {
-            queued_text.push(entity);
+        for (entity, ..) in text_query.iter() {
+            text_queue.push(entity);
         }
```
Comment (konsti219) [inline crates/bevy_ui/src/widget/text.rs:89]:
```suggestion
        text_queue.extend(text_query.iter().map(|(entity, ..)| entity));
```

**Review by konsti219 — CHANGES_REQUESTED** [review]
Using iterators is a bit shorter than a `for` loop

**Comment by ickshonpe** [comment]
I needed to include these changes in #8549, so I might as well close this if #8549 isn't controversial or blocked by anything.

**Pull request closed without merging**